### PR TITLE
Add acceptance test: Verify pulling data from a secret living in another namespace

### PR DESCRIPTION
**Pull request by pratikjagrut**
### Motivation
Verifies GH issue #646 
JIRA TICKET: https://issues.redhat.com/browse/APPSVC-762

### CODE REVIEW

File: test/acceptance/features/bindAppToServiceUsingSecret.feature [hunk 1]
```diff
@@ -394,3 +394,80 @@ Feature: Bind values from a config map referred in backing service resource
         Then Service Binding "ssd-2" is ready
         And The application env var "BACKEND_USERNAME" has value "AzureDiamond"
         And The application env var "BACKEND_PASSWORD" has value "hunter2"
+
+    Scenario: Verify pulling data from a secret living in another namespace
```
Comment (pedjak) [inline test/acceptance/features/bindAppToServiceUsingSecret.feature:398]: the ticket is about verifying, but the scenario name itself should describe the intention, i.e. how system should work from user perspective. Thus:

```
Inject into app all keys from a secret existing in a namespace different from the service resource
```

File: test/acceptance/features/bindAppToServiceUsingSecret.feature [hunk 2]
```diff
@@ -394,3 +394,80 @@ Feature: Bind values from a config map referred in backing service resource
...
+        Binding definition is declared on service CRD.
```
Comment (pedjak) [inline test/acceptance/features/bindAppToServiceUsingSecret.feature:399]: I think this line could be removed.

File: test/acceptance/features/bindAppToServiceUsingSecret.feature [hunk 3]
```diff
@@ -394,3 +394,80 @@ Feature: Bind values from a config map referred in backing service resource
...
+        Given OLM Operator "backend" is running
```
Comment (pedjak) [inline test/acceptance/features/bindAppToServiceUsingSecret.feature:400]: If you use this step, then you should not install CustomResourceDefinition under, or vice versa. I think we need later.

**Review by pedjak — CHANGES_REQUESTED** [review]

File: test/acceptance/features/bindAppToServiceUsingSecret.feature (hunk 1 above)
Comment (pratikjagrut) [reply test/acceptance/features/bindAppToServiceUsingSecret.feature]: Done.

File: test/acceptance/features/bindAppToServiceUsingSecret.feature (hunk 3 above)
Comment (pratikjagrut) [reply test/acceptance/features/bindAppToServiceUsingSecret.feature]: Removed OLM step.

File: test/acceptance/features/bindAppToServiceUsingSecret.feature [hunk 4]
```diff
@@ -394,3 +394,78 @@ Feature: Bind values from a config map referred in backing service resource
...
+                username: AzureDiamond
+                password: hunter2
+            """
+        * The Custom Resource is present
+            """
+            apiVersion: stable.example.com/v1
+            kind: Backend
+            metadata:
+                name: ssa-3-service
+                namespace: backend-services
```
Comment (DhritiShikhar) [inline test/acceptance/features/bindAppToServiceUsingSecret.feature:445]: I can see that service `ssa-3-service` lives in `backend-services` namespace
and secret `ssa-3-secret` lives in `backend-services` namespace.

Both service and secret are in the same namespace... but the title of the PR says that we want to verify a situation where service and secret live in different namespace.

Comment (pratikjagrut) [reply test/acceptance/features/bindAppToServiceUsingSecret.feature:445]: Sorry, @DhritiShikhar I'll update the title service and secret should be in same ns. This test is based on issue #646.

**Review by pedjak — APPROVED** [review]

**Pull request merged**